src/OpenLoco/src/World/CompanyManager.cpp

```diff
@@ -617,6 +617,67 @@ namespace OpenLoco::CompanyManager
         company->expenditures[0][static_cast<uint8_t>(type)] -= payment;
     }
 
+    // 0x0046DD06
+    // id : updatingCompanyId global var
+    // payment : ebp
+    // return : ebp == 0x80000000 for false
+    bool ensureCompanyFunding(const CompanyId id, const currency32_t payment)
```

Comment:
what gameplay action triggers this so i can test it?

Reply:
All commands that cost something should trigger it. But only ai will trigger the loan bit of it